plugins/thehive/komand_thehive/util/api.py

```diff
@@ -89,9 +89,10 @@ def _call_api(
             raise PluginException(preset=PluginException.Preset.UNAUTHORIZED, data=response)
         if response.status_code == 404:
             raise PluginException(preset=PluginException.Preset.NOT_FOUND, data=response)
-        if response.status_code == 500:
+        if response.status_code >= 500:
             raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=response)
         if response.status_code in (200, 201):
             return response.json()
         if response.status_code == 204:
```

Comment:
I would recommend accounting for all 200-299 responses positively as they are all successful and shouldn't raise an error.

Reply:
The previous code we had:
```
if 200 >= response.status_code < 300:
```
caused a lot of issues. 

I'll retry it 👍🏻

Reply:
Yeah so this was the error I was getting when trying to cover whole range with the new Exception that is raised at the end
```
Something unexpected occurred. Check the logs and if the issue persists please contact support. Response was: <Response [201]>
{"body": {"log": "", "status": "error", "meta": {}, "error": "An error occurred during plugin execution!\n\nSomething unexpected occurred. Check the logs and if the issue persists please contact support. Response was: <Response [201]>", "exception": {"cause": "Something unexpected occurred.", "assistance": "Check the logs and if the issue persists please contact support.", "data": "<Response [201]>"}}, "version": "v1", "type": "action_event"}
```

I'm probably just not noticing something really silly here. Log above to keep track ^

Reply:
I would try if response.status_code in range(200, 300)

Reply:
Would it be better to change the ordering so the happy path is checked first to save checking the error codes when it's (hopefully) unlikely to hit these? 

Also is it better to do `data=response.content` in the exception? if we return `Response was <Response 400>` etc it doesn't tell us much but response.content should include the error from the endpoint, the cause/assistance should let us know which status code. might be an issue for `Preset.UNKNOWN` as the exception won't then include the status code, unless an additional logger can be added for that?

Reply:
The order does not matter so much but I've switch them around anyway.
Good catch with the `response.content`. Changing the values now